Sources/App/configure.swift

```diff
@@ -9,7 +10,10 @@ enum ConfigurationError: Error {
 public func configure(_ config: inout Config, _ env: inout Environment, _ services: inout Services) throws {
 
     let logger = PrintLogger()
-    let gitHubEventsService = GitHubEventsService(signatureToken: try Environment.gitHubWebhookSecret())
+    let gitHubEventsService = GitHubEventsService(
+        signatureToken: try Environment.gitHubWebhookSecret(),
+        scheduler: QueueScheduler.main
```

Comment:
The `QueueScheduler.main` always schedule asynchronously, right? (so no risk of deadlocks from that right?)

Also could a solution to investigate that be to run `Wall-E` locally on your Mac and profile it with Instruments to understand what's happening there?

Reply:
Running it locally is the next step if this doesn't work but that would require some configuration and point the webhook to my machine, there's no easy way to debug this end-to-end.